rc/extra/gpg.kak

```diff
@@ -0,0 +1,57 @@
+# Enable kakoune to open, edit, write files encrypted
+# using [GNU privacy guard](https://www.gnupg.org/)
+
+# options {
+declare-option -docstring 'Set this to true to add a signature when saving the file.' bool gpg_sign false
+declare-option -docstring 'Encrypt for this user id. If not set, it is a assumed that a default was configured in ~/.gnupg/gpg.conf.' str gpg_recipient 
+declare-option -docstring 'Set this to a secure deletion command e.g. shred(1).' str gpg_rmcmd 'rm -f'
+# }
+
+hook -group gpg global BufOpenFile .*\.gpg %{
+    set-option buffer autoreload no
+    %sh{
+        f="${kak_buffile%.gpg}"
+        umask 066
+        gpg --output "$f" --yes --decrypt "${kak_buffile}" 2>/dev/null
+        printf %s\\n "
+            evaluate-commands %{
+                edit! '$f'
+                set-option buffer modelinefmt '[GPG] ${kak_opt_modelinefmt}'
+                hook -group gpg buffer BufWritePost %val{buffile} gpg-encrypt
+                hook -group gpg buffer BufClose %val{buffile} gpg-cleanup
+                echo -markup {Information} Decrypted: '${kak_buffile}'
+            }
+        "
+    }
+}
+
+define-command -hidden gpg-encrypt %{
+    %sh{
+        options=""
+        [ -n "${kak_opt_gpg_recipient}" ] && options="-r '${kak_opt_gpg_recipient}'"
+        [ "${kak_opt_gpg_sign}" = "true" ] && options="--sign ${options}"
+
+        cp -f "${kak_buffile}.gpg" "${kak_buffile}.gpg~" 
+        eval gpg "${options}" --yes --encrypt "${kak_buffile}"
+        printf %s\\n "
+            echo -markup {Information} Encrypted: ${kak_buffile}.gpg
+        "
+    }
+
+}
+
+define-command -hidden gpg-cleanup  %{
+    %sh{
+        if [ -f "${kak_buffile}" ] && [ -n "${kak_opt_gpg_rmcmd}" ]; then
+            # Using shred(1) is naturally very slow for large files because it has
+            # to overwrite the whole temp file with zeros. Improve kakoune's
+            # responsiveness upon closing large files by forking off the
+            # deletion process.
+            (eval "${kak_opt_gpg_rmcmd} ${kak_buffile}") </dev/null >/dev/null 2>&1 &
```

Comment:
Braces `{`/`}` could be used here instead of parenthesis to prevent a subprocess to be spawn.

Reply:
When I use braces it doesn't parse in Kakoune. :disappointed: